db/db_impl/db_impl_secondary.cc

```diff
@@ -17,8 +17,9 @@ namespace ROCKSDB_NAMESPACE {
 
 #ifndef ROCKSDB_LITE
 DBImplSecondary::DBImplSecondary(const DBOptions& db_options,
-                                 const std::string& dbname)
-    : DBImpl(db_options, dbname) {
+                                 const std::string& dbname,
+                                 const std::string& secondary_path)
+    : DBImpl(db_options, dbname), secondary_path_(secondary_path) {
```

Comment:
(unrelated to this PR) I think `DBImpl` should be passed `read_only=true`. That's a new flag introduced in #8142 to prevent `SanitizeOptions()` from creating an info logger when none is provided (i.e., `info_log == nullptr`). I'd guess in a secondary DB, a `SanitizeOptions()`-created info logger could conflict with the primary DB info logger. It looks like `OpenAsSecondary()` tries to avoid opening the DB with option `info_log == nullptr` -- except when  L616 is executed.

Reply:
Thanks for pointing it out, I need look into it to understand the problem, filed an issue to myself: #8193